source/extensions/quic_listeners/quiche/platform/quic_pcc_sender_impl.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "quiche/quic/core/quic_types.h"
+
+namespace quic {
+
+class QuicClock;
+class QuicConnectionStats;
+class QuicRandom;
+class QuicUnackedPacketMap;
+class RttStats;
+class SendAlgorithmInterface;
+
+// Interface for creating a PCC SendAlgorithmInterface.
+SendAlgorithmInterface* CreatePccSenderImpl(const QuicClock* clock, const RttStats* rtt_stats,
+                                            const QuicUnackedPacketMap* unacked_packets,
+                                            QuicRandom* random, QuicConnectionStats* stats,
+                                            QuicPacketCount initial_congestion_window,
+                                            QuicPacketCount max_congestion_window) {
+  return nullptr;
```

Comment:
How about using `NOT_REACHED_GCOVR_EXCL_LINE` instead of implicitly return nullptr?

Reply:
From source/common/common/assert.h:
// NOT_REACHED_GCOVR_EXCL_LINE is for spots the compiler insists on having a return, but where we
// know that it shouldn't be possible to arrive there, assuming no horrendous bugs.

In this case, this line is actually reachable in test, so I added a PANIC instead.

Reply:
SGTM